File: UserControlForm/UserControlForm/UserControlForm.Web/Modules/ServerTypes/Administration/UserFormEditorColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { UserFormEditorRow } from "./UserFormEditorRow";

export interface UserFormEditorColumns {
    TemplateId: Column<UserFormEditorRow>;
    FormName: Column<UserFormEditorRow>;
    CreatedByUsername: Column<UserFormEditorRow>;
    CreatedDate: Column<UserFormEditorRow>;
    ModifiedByUsername: Column<UserFormEditorRow>;
    ModifiedDate: Column<UserFormEditorRow>;
}

export class UserFormEditorColumns extends ColumnsBase<UserFormEditorRow> {
    static readonly columnsKey = 'Administration.UserFormEditor';
    static readonly Fields = fieldsProxy<UserFormEditorColumns>();
}

File: UserControlForm/UserControlForm/UserControlForm.Web/Modules/Administration/UserFormEditor/UserFormEditorColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace UserControlForm.Administration.Columns;

[ColumnsScript("Administration.UserFormEditor")]
[BasedOnRow(typeof(UserFormEditorRow), CheckNames = true)]
public class UserFormEditorColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int TemplateId { get; set; }
    
    [EditLink]
    public string FormName { get; set; }
    
    public string CreatedByUsername { get; set; }
    
    public DateTime CreatedDate { get; set; }
    
    public string ModifiedByUsername { get; set; }
    
    public DateTime ModifiedDate { get; set; }
}